### PR TITLE
DOC: optimize: update Powell docs to reflect API

### PR DESCRIPTION
**Pull request by jonas-eschle**
Fixes that the docs would advertise a "bounds" option for Powell (https://docs.scipy.org/doc/scipy/reference/optimize.minimize-powell.html#optimize-minimize-powell) while this is the same as given to minimize (https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html). Since the `options` are passed through unpacked, specifying both will fail.

Clarified this fact in the docs.

### CODE REVIEW

File: scipy/optimize/_hessian_update_strategy.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def update(self, delta_x, delta_grad):
             if self.init_scale == "auto":
                 scale = self._auto_scale(delta_x, delta_grad)
             else:
-                scale = float(self.init_scale)
+                scale = self.init_scale
```
Comment (tylerjereddy) [inline scipy/optimize/_hessian_update_strategy.py:193]: The CI failure looks related--may be good to update the PR/description to clarify that the change is extending beyond documentation only now.

**Comment by mayou36** [comment]
My fault, I managed to put in things here that don't belong, sorry!

Reverted the wrong commit, now it is what it says :)

**Comment by tupui** [comment]
Thanks for noticing this. But we would have the same for all methods accepting bounds no? (like `_minimize_lbfgsb`). I would prefer to handle this on the `minimize` side instead for everyone. So just checking for conflicts between all parameters passed to the functions.

**Comment by mayou36** [comment]
So the problem is currently only present in Powell as others don't describe the bounds as a valid parameter. In this sense, this is a "bug", while the others are just "missing docs of what they do exactly".

About how to fix it in general...

IMHO - and that is really just a personal opinion on a specific design choice (I understand implications of such a large library and also that not every design choice (or none at all) are obvious at the beginning and I am of course extremely grateful for SciPy and all the work you do!) - the whole unified API approach is an inherently flawed approach. While the minimization call of a function with the initial values should always be the same, the minimization functions are so different by themselves that the API got super cluttered and things like this (or #13524, #13557) are just bound to happen as we have nparams times n optimizer choices. In order to separate the two, https://github.com/scipy/scipy/pull/8414 seems to be the solution, which is stalled and seems unlikely to be merged or even approved.

For this problem here: explaining the bounds in each minimizer page makes more sense to me as the handling is minimizer specific. Why should it be written in the main minimizer what a specific minimizer does with the bounds if I only care about one specific minimizer? Given though that it is currently solved this way (and it means that one needs to read the whole docs just to get a singe piece of information for one specific minimizer), so doing it like this would not fall out of line.

But then I don't mind at all, as said, I would completely refactor it, just fixing some patches here. What do you prefer? Should I remove the entry of bounds completely (like for the others) and add it in the main docs?

**Comment by tupui** [comment]
@andyfaff any opinion on this one?

For now, I would just update the doc for now to be consistant with the other methods.

But #8414 (or something else) seems needed at some point.

**Comment by mayou36** [comment]
Also looking at #13754 I think the only way forward is to declutter the docs and put them where they belong, which is to the minimizer as in the PR. This follows the principle that arguments are maybe common but better explained, or what it does for the minimizer, at the actual docs page.

**Comment by tupui** [comment]
@mayou36 I cross-referenced this as we are going to have soon the same issue with Nelder-Mead.

As I said over the other PR, this should be fixed in the `minimize` side (either in the doc or before unpacking and passing to the optimizers).

I agree with you that some options are missing, although this might be intended as it's necessary to not re-declare options for `scipy.optimize.show_options` to be not be redundant with parameters. There is definitely something to clarify here.

As for this PR, as it is IMO, I would not do this. We should not describe in the docstrings something that might happen because of another function wrongly calling this one. The current docstrings does describe accurately the parameters.

I would then suggest to change your PR to change `scipy/optimize/_minimize.py:minimize`. As for which of the 2 solutions (or something else), let's wait for @andyfaff.

**Comment by mayou36** [comment]
I agree; as mentioned this was intended as a simple ad-hoc fix, so anything else is fine oc. Sure, let's see

**Comment by tupui** [comment]
@andyfaff any update on this set of issues?

**Comment by tupui** [comment]
Friendly reminder @andyfaff

**Comment by mdhaber** [comment]
In the method-specific documentation of `minimize`, there is only space for documenting `options`:
image

but in the method-specific documentation of `minimize`, we (re-) document all the parameters of `linprog` 
image

and then separately, all the `options`. 
image

This way, we are able to specify what `linprog(method='highs')` will do with `bounds` (which could be different than the other `method`s), but `bounds` is not listed as an "Option", so there is no reason that the user would think it would be valid to include it within the `options` dictionary. How does that sound?

File: scipy/optimize/_optimize.py [hunk 2]
```diff
@@ -3190,17 +3254,6 @@ def _minimize_powell(func, x0, args=(), callback=None, bounds=None,
...
-        within the bounds. If bounds are provided, the initial guess is outside
-        the bounds, and `direc` is full rank (or left to default), then some
-        function evaluations during the first iteration may be outside the
-        bounds, but every function evaluation after the first iteration will be
-        within the bounds. If `direc` is not full rank, then some parameters may
-        not be optimized and the solution is not guaranteed to be within the
-        bounds.
     return_all : bool, optional
         Set to True to return a list of the best solution at each of the
         iterations.
```
Comment (mdhaber) [inline scipy/optimize/_optimize.py:3259]:
```suggestion
```

File: scipy/optimize/_optimize.py [hunk 3]
```diff
@@ -3190,20 +3254,6 @@ def _minimize_powell(func, x0, args=(), callback=None, bounds=None,
...
-        within the bounds. If bounds are provided, the initial guess is outside
-        the bounds, and `direc` is full rank (or left to default), then some
-        function evaluations during the first iteration may be outside the
-        bounds, but every function evaluation after the first iteration will be
-        within the bounds. If `direc` is not full rank, then some parameters may
-        not be optimized and the solution is not guaranteed to be within the
-        bounds.
-    return_all : bool, optional
-        Set to True to return a list of the best solution at each of the
-        iterations.
```
Comment (mdhaber) [inline scipy/optimize/_optimize.py:3206]: This was redundant with `return_all` above.

**Comment by mdhaber** [comment]
@andyfaff @tupui this was a pretty simple doc-only change, so I went ahead and merged it.

**Comment by tupui** [comment]
Thanks @mdhaber for taking this up and @jonas-eschle for the PR 😃

**Pull request merged**